code/datums/components/mood.dm

```diff
@@ -212,6 +213,16 @@
 	qdel(event)
 	update_mood()
 
+/datum/component/mood/proc/remove_temp_moods(var/admin) //Removes all temp moods
+	for(var/i in mood_events)
+		var/datum/mood_event/moodlet = mood_events[i]
+		if(!moodlet || !moodlet.timeout)
+			return 0
```

Comment:
Seems like this will not remove temporary moodlets which are after a permanent moodlet in the list

Reply:
wow yeah I'm dumb, thanks for pointing that one out.